src/Compilers/Core/Portable/InternalUtilities/FatalError.cs

```diff
@@ -11,6 +11,18 @@
 
 namespace Microsoft.CodeAnalysis.ErrorReporting
 {
+    /// <summary>
+    /// Thrown when async code must cancel the current execution but does not have access to the <see cref="CancellationTokenSource"/> of the <see cref="CancellationToken"/> passed to the code.
+    /// Should be used in very rare cases where the <see cref="CancellationTokenSource"/> is out of our control (e.g. owned but not exposed by JSON RPC in certain call-back scenarios).
+    /// </summary>
+    internal sealed class OperationCanceledNotMatchingCancellationTokenException : OperationCanceledException
```

Comment:
💡 I found this exception name misleading, since it (to me) suggests we check for matching cancellation tokens. Perhaps `OperationCanceledIgnoringCallerTokenException`?

Reply:
Renamed.